### PR TITLE
feature(enhancement): Blueshift update group call

### PR DESCRIPTION
**Pull request by sauravlal-rudderstack**
## Description of the change

> Update Group call.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/blueshift/transform.js [hunk 1]
```diff
@@ -116,58 +113,29 @@ const identifyResponseBuilder = async (message, category, { Config }) => {
...
-      }
-    } else {
-      throw new CustomError(
-        "[Blueshift]:: name and description are required to create an empty list, or List Id is required to add user.",
-        400
-      );
-    }
-  }
+  payload.event = "identify";
+  payload = extractCustomFields(message, payload, ["traits"], ["cookie"]);
```
Comment (shrouti1507) [inline v0/destinations/blueshift/transform.js:130]: Suggestion: As `cookie` is being mapped from traits only and it is **not** a mandatory field, we can simply move all the properties from traits here and do not need to add it in the mapping json.

File: v0/destinations/blueshift/data/blueshiftGroupConfig.json [hunk 2]
```diff
@@ -1,47 +1,85 @@
...
-    "required": false
+    "required": true
   },
   {
-    "destKey": "identifier_value",
+    "destKey": "customer_id",
     "sourceKeys": "userId",
     "sourceFromGenericMap": true,
-    "required": true
+    "required": false
```
Comment (utsabc) [inline v0/destinations/blueshift/data/blueshiftGroupConfig.json:12]:
```suggestion
    "required": true
```
Isn't userId mandatory for mapping?

File: v0/destinations/blueshift/data/blueshiftGroupConfig.json [hunk 3]
```diff
@@ -1,47 +1,85 @@
...
+    "sourceKeys": ["context.idfv", "context.device.id"],
+    "required": false
+  },
+  {
+    "destKey": "device_manufacturer",
+    "sourceKeys": "context.device.manufacturer",
+    "required": false
+  },
+  {
+    "destKey": "os_name",
```
Comment (utsabc) [inline v0/destinations/blueshift/data/blueshiftGroupConfig.json:51]: Are these fields populating for each user in group ?

File: v0/destinations/blueshift/transform.js [hunk 4]
```diff
@@ -116,58 +113,29 @@ const identifyResponseBuilder = async (message, category, { Config }) => {
...
+  if (!Config.eventApiKey) {
     throw new CustomError(
-      "[BLUESHIFT] User API Key required for Authentication.",
+      "[BLUESHIFT] event API Key required for Authentication.",
       400
     );
   }
 
-  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  let payload = constructPayload(message, MAPPING_CONFIG[category.name]);
```
Comment (utsabc) [inline v0/destinations/blueshift/transform.js:123]:
```suggestion
  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
```

**Comment by utsabc** [comment]
We have removed quite a number of integration tests are we covering all the scenarios with unit test cases?

File: v0/destinations/blueshift/transform.js (hunk 1 above)
Comment (utsabc) [reply v0/destinations/blueshift/transform.js:130]: In mapping JSON the mapping is sourced from specific fields like `context.device.id` which will be difficult to implement using this function

**Comment by sauravlal-rudderstack** [comment]
> We have removed quite a number of integration tests are we covering all the scenarios with unit test cases?

> Yeah, previously we have lots of conditions so built a test case accordingly, but now we change, so it has no use.

File: v0/destinations/blueshift/transform.js (hunk 4 above)
Comment (utsabc) [reply v0/destinations/blueshift/transform.js:123]: You cannot reassign `const` but you can modify property of a `const` variable
Example 
```
const payload = { key: value };
payload.key2 = value
```
The above is allowed

```
const payload = { key: value };
payload  = { key2: value2 }
```
The above is not allowed


```
let payload = { key: value };
payload  = { key2: value2 }
```
The above is allowed

File: v0/destinations/blueshift/data/blueshiftGroupConfig.json (hunk 2 above)
Comment (utsabc) [reply v0/destinations/blueshift/data/blueshiftGroupConfig.json:12]: So sending a group call to blueshift without `customer_id` works?

Comment (utsabc) [reply v0/destinations/blueshift/data/blueshiftGroupConfig.json:12]: How is the particular `user` identified then ?

**Review by utsabc — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Comment by utsabc** [comment]
Need to check `eu` instead of `EU` in transformer

**Review by utsabc — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by akashrpo — APPROVED** [review]

**Pull request merged**